### DISCUSSION TITLE
Mainline Merge from Hotfix to main does not increment version with custom merge commit message

### DISCUSSION THREAD

**Discussion by sharpSteff** [GitTools/GitVersion#3649] [Q&A] [answered]
**Describe the bug**
I have an main (1.0.0) and hotfix (1.0.1) branch. I don't have anything fancy in my GitVersion.yml

The main branch stays at 1.0.0, when merging the hotfix branch with a merge custom message via 
```
$ git merge --no-ff hotfix/linux_sdcard -m "Merge pull request '[Fix] #12344 - Foobar' (#3) from hotfix/linux into release

Reviewed-on: https://XXXXXXXXX/git/XXXXXXXX/pulls/3"
```

When merging without a custom message the version increments to 1.0.1 on main:
`$git merge --no-ff hotfix/test`

I did observe the issue because GITEA merges the PR with following message:
```
Merge pull request '[Fix] #12344 - Foobar' (#3) from hotfix/linux into release

Reviewed-on: https://XXXXXXXXX/git/XXXXXXXX/pulls/3
```

## Expected Behavior
The version at main increments regardless of a message merge message

## Actual Behavior
The version at main does not increment when mergen with a custom merge message

## Steps to Reproduce
Clone https://github.com/sharpSteff/gitversion_bug

`$ git checkout release`
```
$ git merge --no-ff hotfix/linux_sdcard -m "Merge pull request '[Fix] #12344 - Foobar' (#3) from hotfix/linux into release

Reviewed-on: https://XXXXXXXXX/git/XXXXXXXX/pulls/3"
```
`$ dotnet-gitversion`

- Version Used: 5.12
- Operating System and version (Windows 10):

**Comment by HHobeck** [comment]
I'm going to move this issue to the QA section because it seems to me not a bug.

**Comment by HHobeck** [comment, marked as answer]
Hi there. ;)

> Expected Behavior
The version at main increments regardless of a message merge message

If you think about it: This expecation is unrealstic. How would you determine the information where the merge comes from without putting it in the commit message?

If you have custom merge message format you need to specify it in `GitVersion.yml` configuration. The documentation says:

> ### merge-message-formats
> Custom merge message formats to enable identification of merge messages that do not
> follow the built-in conventions.  Entries should be added as key-value pairs where
> the value is a regular expression.
> e.g.
> ```yaml
> merge-message-formats:
>     tfs: '^Merged (?:PR (?<PullRequestNumber>\d+)): Merge (?<SourceBranch>.+) to (?<TargetBranch>.+)'
> ```

**Reply by sharpSteff** [reply]
@HHobeck 
thanks for the quick response.
I'll try your solution. 

this topic probably also happens for github-repos as well ,doesn't  it? what's the solution for them? There has to be some predefined settings. 

you are right, gitversion has to check the commits, since there is no branch information in the git-commit-tree.

**Comment by sharpSteff** [comment]
For onyone using gitea with gitversion:

this is my merge-message-formats:
```
merge-message-formats:
    gitea: '^Merge pull request .*\(!(?<PullRequestNumber>\d+)\) from (?<SourceBranch>.+) into (?<TargetBranch>.+)'
```

**Reply by HHobeck** [reply]
You can find the default regular expression here: https://github.com/GitTools/GitVersion/blob/main/src/GitVersion.Core/MergeMessage.cs#L9

Your regular expression looks like `GitHubPull` which is already included:
`^Merge pull request #(?<PullRequestNumber>\d+) (from|in) (?:[^\s\/]+\/)?(?<SourceBranch>[^\s]*)(?: into (?<TargetBranch>[^\s]*))*`
